aws/signer/v4/v4.go

```diff
@@ -321,7 +331,7 @@ func (v4 Signer) signWithBody(r *http.Request, body io.ReadSeeker, service, regi
 	// If the request is not presigned the body should be attached to it. This
 	// prevents the confusion of wanting to send a signed request without
 	// the body the request was signed for attached.
-	if !ctx.isPresign {
+	if !v4.DisableRequestBodyOverwrite && !ctx.isPresign {
```

Comment:
`!(v4.DisableRequestBodyOverwrite || ctx.isPresign)`

Don't know if the Go compiler does De Morgan's laws under the hood.